File: cs/Minsk/CodeAnalysis/Binding/BoundVariableExpression.cs
namespace Minsk.CodeAnalysis.Binding;

internal sealed class BoundVariableExpression : BoundExpression
{
    public BoundVariableExpression(VariableSymbol variable)
    {
        Variable = variable;
    }

    public VariableSymbol Variable { get; }
    public override BoundNodeKind Kind => BoundNodeKind.VariableExpression;
    public override Type Type => Variable.Type;
}

File: nim/src/minsk/codeAnalysis/binding/expressions/boundVariableExpression.nim
import minsk/minskObject
import minsk/codeAnalysis/binding/[
  boundExpression,
  boundNodeKind,
]
import minsk/codeAnalysis/variableSymbol

type
  BoundVariableExpression* = ref object of BoundExpression
    variable*: VariableSymbol

proc newBoundVariableExpression*(variable: VariableSymbol): BoundVariableExpression =
  new(result)
  result.variable = variable

method kind*(self: BoundVariableExpression): BoundNodeKind =
  BoundNodeKind.VariableExpression

method ty*(self: BoundVariableExpression): MinskObjectKind =
  self.variable.ty
